database.cc

```diff
@@ -1387,7 +1387,6 @@ database::query(schema_ptr s, const query::read_command& cmd, query::result_opti
     }
     column_family& cf = find_column_family(cmd.cf_id);
     auto& semaphore = get_reader_concurrency_semaphore();
-    auto class_config = query::query_class_config{.semaphore = semaphore, .max_memory_for_unlimited_query = *cmd.max_result_size};
```

Comment:
Indeed this struct has no reason to exist anymore:
* `max_memory_for_unlimited_query` is always equal to `cmd.max_result_size` since this comes from the coordinator.
* The semaphore is unused since we moved admission in front of the cache.
Please also remove the struct itself, I don't think there are any more users.